scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala

```diff
@@ -495,6 +500,103 @@ object TableRowJsonIO {
   )
 }
 
+final case class BigQueryPartitionedTable[T: Coder](writer: beam.BigQueryIO.Write[T])(
```

Comment:
I wonder if we could just tweak `BigQueryTypedTable` a little bit.
Currently it looks like:

https://github.com/spotify/scio/blob/d387eeb901babd091b83ac5630eb793f9be2416d/scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala#L324-L329

and:

https://github.com/spotify/scio/blob/d387eeb901babd091b83ac5630eb793f9be2416d/scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala#L341

I think we could just not assume that `Table` is always defined and move this logic to the companion object. This way we can have an extra method that accepts a `DynamicDestination` or a `ValueInSingleWindow[T] => TableDestination`.

We would avoid some duplication and reuse a lot of the existing logic.

Reply:
Hmm, I see your point but `BigQueryTypedTable` also supports `read` and can be used in [test context](
https://github.com/spotify/scio/blob/e5708e62111f30dd34fff39d851ea97cb42ff017/scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/dynamic/syntax/SCollectionSyntax.scala#L74:L77). These weren't supported in a `dynamic/syntax/SCollectionSyntaz`. 

Should a partitioned table support these as well?

Reply:
right now we don't have a really good way to test it so we can keep it as is. However, the idea of tweaking `BigQueryTypeTable` was so you could use it as underlying impl ([example](https://github.com/spotify/scio/tree/master/scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala)) for the `BigQueryPartionedTable`. On a side note we should also rename it to smth like `BigQueryDynamicTable`.

Reply:
👍  Addressed this in the latest commits